yt_dlp/extractor/radiocomercial.py

```diff
@@ -0,0 +1,141 @@
+import re
+from collections import namedtuple
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+)
+
+
+class RadioComercialBaseExtractor(InfoExtractor):
+    def _extract_page_content(self, url):
+        video_id = RadioComercialIE._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title')
+        url = self._html_search_regex(r'<a.+?isExclusivePlay=.+?href="(.+?)">', webpage, 'url')
+        date = self._html_search_regex(r'<div[^"]+"date">(\d{4}-\d{2}-\d{2})</div>', webpage, 'date')
+        thumbnail = self._html_search_regex(r'<source[^"]+"image/jpeg[^/]+(.+?)\">', webpage, 'thumbnail')
+        season = int_or_none(self._html_search_regex(r'<h2>\w+\s(\d+)</h2>', webpage, 'season'))
+        episode_id = int_or_none(self._html_search_regex(r'episodeid=(\d+)&', url, 'episode_id'))
+
+        return {
+            'id': video_id,
+            'title': title,
+            'date': date,
+            'thumbnail': thumbnail,
+            'season': season,
+            'episode_id': episode_id,
+            'url': url
+        }
+
+
+class RadioComercialIE(RadioComercialBaseExtractor):
+    _VALID_URL = r'https?://(?:www\.)?radiocomercial\.pt/podcasts/[^/]+/\w\d+/(?P<id>[-\w]+)/*$'
+    _TESTS = [{
+        'url': 'https://radiocomercial.pt/podcasts/o-homem-que-mordeu-o-cao/t6/taylor-swift-entranhando-se-que-nem-uma-espada-no-ventre-dos-fas',
+        'md5': '5f4fe8e485b29d2e8fd495605bc2c7e4',
+        'info_dict': {
+            'id': 'taylor-swift-entranhando-se-que-nem-uma-espada-no-ventre-dos-fas',
+            'ext': 'mp3',
+            'title': 'Taylor Swift entranhando-se que nem uma espada no ventre dos fãs.',
+            'date': '2023-10-25',
+            'thumbnail': r're:/upload/[^.]+.jpg',
+            'season': 6,
+            'episode_id': 220899
+        }
+    },
+        {
+        'url': 'https://radiocomercial.pt/podcasts/convenca-me-num-minuto/t3/convenca-me-num-minuto-que-os-lobisomens-existem',
+        'md5': '47e96c273aef96a8eb160cd6cf46d782',
+        'info_dict': {
+            'id': 'convenca-me-num-minuto-que-os-lobisomens-existem',
+            'ext': 'mp3',
+            'title': 'Convença-me num minuto que os lobisomens existem',
+            'date': '2023-10-26',
+            'thumbnail': r're:/upload/[^.]+.jpg',
+            'season': 3,
+            'episode_id': 221210
+        }
+    },
+    ]
+
+    def _real_extract(self, url):
+        return self._extract_page_content(url)
+
+
+class RadioComercialPlaylistIE(RadioComercialBaseExtractor):
+    _VALID_URL = r'https?://(?:www\.)?radiocomercial\.pt/podcasts/(?P<id>[-\w]+)[/\w\d+]*$'
```

Comment:
The returned id should reflect which season is being downloaded. Otherwise, different seasons would end up with the same id.

Reply:
I'm not sure I understood your suggestion @seproDev . Could you please provide me an example?

Reply:
So your suggestion would be to do something like this?
`'id': f'temporada{season}_{video_id}',`

Reply:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?radiocomercial\.pt/podcasts/(?P<id>[\w-]+)(?:/t?(?P<season>\d+))?'
```